internal/controller.go

```diff
@@ -182,6 +182,18 @@ func (c *Controller) GetWorkerPool(ctx context.Context) (out *WorkerPool, err er
 			return err
 		}
 
+		// Remove any drained workers from the list of workers in the pool.
+		// These don't count towards the "available workers", so they shouldn't
+		// be included when making a scaling decision.
+		worker_index := 0
+		for _, worker := range wpDetails.Pool.Workers {
+			if !worker.Drained {
+				wpDetails.Pool.Workers[worker_index] = worker
+				worker_index++
+			}
+		}
+		wpDetails.Pool.Workers = wpDetails.Pool.Workers[:worker_index]
```

Comment:
Not sure if there's a better way to do this (I'm not particularly experienced with Golang), but this _does_ work in tests and in practice.

Reply:
Decided to remove the entries in-place to avoid needing to modify any other code.